flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/join/TemporalRowtimeJoin.scala

```diff
@@ -205,6 +203,7 @@ class TemporalRowtimeJoin(
     val rightRowsSorted = getRightRowsSorted(rightRowtimeComparator)
     var lastUnprocessedTime = Long.MaxValue
 
+    // TODO: here we should check if the right side has anything or not (optimization).
```

Comment:
@pnowojski What is your opinion on this one?

Reply:
You can create a separate hotfix for that if you want. It's a valid and simply improvement, but probably not super critical.